worlds/oot_soh/location_access/overworld/market.py

```diff
@@ -5,8 +5,272 @@
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT1, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT2, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT3, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT4, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT5, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT6, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT7, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT8, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT9, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT10, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT11, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE1, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE2, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE3, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE4, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE5, lambda bundle: (is_child(bundle) and can_break_crates(bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_GUARD_HOUSE, world, [
+        (Regions.MARKET_ENTRANCE, lambda bundle: True)
+    ])
+    
+    ## Market Bazaar
+    # Locations
+    add_locations(Regions.MARKET_BAZAAR, world, [
+        (Locations.MARKET_BAZAAR_ITEM1, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM2, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM3, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM4, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM5, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM6, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM7, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM8, lambda bundle: True),
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_BAZAAR, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Mask Shop
+    # Events
+    add_events(Regions.MARKET_MASK_SHOP, world, [
+        # Todo: Is this accurate? CAN_BORROW_MASKS,BORROW_SKULL_MASK in Events, rest missing. Add to Enums?
```

Comment:
I've added some events on my side for borrowing all the masks. Use these for your events here.
<img width="402" height="84" alt="image" src="https://github.com/user-attachments/assets/afccfdd2-947b-439d-85f0-a0da2c9ead1f" />